public/js/utils/redux/middleware/promise.js

```diff
@@ -19,34 +20,35 @@ function promiseMiddleware({ dispatch, getState }) {
       return next(action);
     }
 
-    const promiseInst = action[PROMISE];
-    const seqId = seqIdGen().toString();
-
-    // Create a new action that doesn't have the promise field and has
+    // Mutate the action, removing the promise field and adding
     // the `seqId` field that represents the sequence id
-    action = Object.assign(
-      toObject(entries(action).filter(pair => pair[0] !== PROMISE)), { seqId }
-    );
-
-    dispatch(Object.assign({}, action, { status: "start" }));
+    const { [PROMISE]: promiseInst, ...actionWithoutPromise } = action;
+    action = {
+      ...actionWithoutPromise,
+      seqId: seqIdGen().toString(),
+      status: "start",
+    };
+    dispatch(action);
```

Comment:
These are fine changes, but let's make every dispatch the same: it needs to add the appropriate async props related to that action:

```
const { [PROMISE]: promiseInst, ...restProps } = action;
action = { ...restProps, seqIdGen().toString() };

dispatch({ ...action, status: "start" });
```

It's very clear this way what's going on: there are 3 dispatches in this file and each of them are dispatching start/done/error actions.